var/spack/repos/builtin/packages/mvapich/package.py

```diff
@@ -0,0 +1,300 @@
+    #   single      - No threads (MPI_THREAD_SINGLE)
+    #   funneled    - Only the main thread calls MPI (MPI_THREAD_FUNNELED)
+    #   serialized  - User serializes calls to MPI (MPI_THREAD_SERIALIZED)
+    #   multiple    - Fully multi-threaded (MPI_THREAD_MULTIPLE)
+    #   runtime     - Alias to "multiple"
+    variant(
+        "threads",
+        default="multiple",
+        values=("single", "funneled", "serialized", "multiple"),
+        multi=False,
+        description="Control the level of thread support",
+    )
+
+    # 32 is needed when job size exceeds 32768 cores
+    variant(
+        "ch3_rank_bits",
+        default="32",
+        values=("16", "32"),
+        multi=False,
+        description="Number of bits allocated to the rank field (16 or 32)",
+    )
+
+    variant(
+        "process_managers",
+        description="List of the process managers to activate",
+        values=disjoint_sets(("auto",), ("slurm",), ("hydra", "gforker", "remshell"))
+        .prohibit_empty_set()
+        .with_error("'slurm' or 'auto' cannot be activated along with " "other process managers")
+        .with_default("auto")
+        .with_non_feature_values("auto"),
+    )
+
+    variant(
+        "netmod",
+        description="Select the netmod to be enabled for this build."
+        "For IB/RoCE systems, use the ucx netmod, for interconnects supported "
+        "by libfabrics, use the ofi netmod. For more info, visit the "
+        "homepage url.",
+        default="ofi",
+        values=(
+            "ofi",
+            "ucx",
+            #       "ucr",
+        ),
+        multi=False,
+    )
+
+    variant(
+        "alloca", default=False, description="Use alloca to allocate temporary memory if available"
+    )
+
+    variant(
+        "file_systems",
+        description="List of the ROMIO file systems to activate",
+        values=auto_or_any_combination_of("lustre", "gpfs", "nfs", "ufs"),
+    )
+
+    depends_on("findutils", type="build")
+    depends_on("bison", type="build")
+    depends_on("pkgconfig", type="build")
+    depends_on("zlib")
+    depends_on("libpciaccess", when=(sys.platform != "darwin"))
+    depends_on("libxml2")
+    depends_on("cuda", when="+cuda")
+    # depends_on("rdma-core", when="netmod=ucr")
+    depends_on("libfabric", when="netmod=ofi")
+    depends_on("slurm", when="process_managers=slurm")
+    depends_on("ucx", when="netmod=ucx")
+
+    filter_compiler_wrappers("mpicc", "mpicxx", "mpif77", "mpif90", "mpifort", relative_root="bin")
+
+    @classmethod
+    def determine_version(cls, exe):
+        if exe.endswith("mpichversion"):
+            output = Executable(exe)(output=str, error=str)
+            match = re.search(r"^MVAPICH2 Version:\s*(\S+)", output)
+        elif exe.endswith("mpiname"):
+            output = Executable(exe)("-a", output=str, error=str)
+            match = re.search(r"^MVAPICH2 (\S+)", output)
+        return match.group(1) if match else None
```

Comment:
This I think needs to change, or the same executable will be discovered by both `mvapich2` and `mvapich`

Reply:
For the 3.0a version we have not yet changed this part.  We will change this for the next release.  For now we would like to keep this section in.